src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +38,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
```

Comment:
Delegate validating address to the different component classes you have created now. That is, Block should if the block number is formatted correctly.